pkg/status/status.go

```diff
@@ -365,27 +389,24 @@ func (s *status) monitorClusterStatus() {
 				break
 			}
 
-			// Report that marketplace is available after meeting minimal syncs.
-			if cohelpers.IsStatusConditionFalse(s.clusterOperator.Status.Conditions, configv1.OperatorAvailable) {
-				reason := "OperatorAvailable"
-				conditionListBuilder := clusterStatusListBuilder()
-				conditionListBuilder(configv1.OperatorProgressing, configv1.ConditionFalse, fmt.Sprintf("Successfully progressed to release version: %s", s.version), reason)
-				conditionListBuilder(configv1.OperatorUpgradeable, configv1.ConditionTrue, upgradeableMessage, reason)
-				statusConditions := conditionListBuilder(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Available release version: %s", s.version), reason)
-				statusErr = s.setStatus(statusConditions)
-				break
-			}
-
 			// Update the status with the appropriate state.
 			isSucceeding, ratio := s.syncRatio.IsSucceeding()
 			if ratio != nil {
 				var statusConditions []configv1.ClusterOperatorStatusCondition
 				conditionListBuilder := clusterStatusListBuilder()
-				conditionListBuilder(configv1.OperatorUpgradeable, configv1.ConditionTrue, upgradeableMessage, "OperatorAvailable")
-				if isSucceeding {
-					statusConditions = conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionFalse, fmt.Sprintf("Current CR sync ratio (%g) meets the expected success ratio (%g)", *ratio, successRatio), "OperandTransitionsSucceeding")
+				reason := "OperatorAvailable"
+				conditionListBuilder(configv1.OperatorProgressing, configv1.ConditionFalse, fmt.Sprintf("Successfully progressed to release version: %s", s.version), reason)
+				conditionListBuilder(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Available release version: %s", s.version), reason)
+				conditionListBuilder(configv1.OperatorUpgradeable, configv1.ConditionTrue, upgradeableMessage, reason)
+				defaultOpsrcName, err := getFirstKeyValue(s.defaultOpsrcErrors)
+				if defaultOpsrcName != "" {
+					statusConditions = conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionTrue, fmt.Sprintf("Default OperatorSource (%s) is failing: %v", defaultOpsrcName, err.Error()), err.Reason())
```

Comment:
Is there a reason we wouldn't want to see _all_ of the failing default operatorsources here?

Reply:
No - but we can only report a single condition that summaries why marketplace has degraded. Furthermore, the only information retrieved via telemetry includes the Name, Status, and Reason. Reason has to be a PascalCase string and I couldn't find a way to report multiple `opsrc` failures with such a requirement. Thoughts?

Reply:
Convinced in person